src/plugins/condition/criterion/AllTelemetryCriterion.js

```diff
@@ -127,20 +128,37 @@ export default class AllTelemetryCriterion extends TelemetryCriterion {
                 options
             ));
 
+        let telemetryDataCache = {};
         return Promise.all(telemetryRequests)
             .then(telemetryRequestsResults => {
                 let latestDatum;
+                let latestTimestamp;
+                const timeSystems = this.openmct.time.getAllTimeSystems();
+                const timeSystem = this.openmct.time.timeSystem();
+
                 telemetryRequestsResults.forEach((results, index) => {
                     latestDatum = results.length ? results[results.length - 1] : {};
-                    if (index < telemetryRequestsResults.length-1) {
-                        if (latestDatum) {
-                            this.telemetryDataCache[latestDatum.id] = this.computeResult(latestDatum);
-                        }
+                    if (latestDatum) {
+                        latestDatum.id = keys[index];
+                        telemetryDataCache[latestDatum.id] = this.computeResult(latestDatum);
+
+                        latestTimestamp = getLatestTimestamp(
+                            latestTimestamp,
+                            latestDatum,
+                            timeSystems,
+                            timeSystem
+                        );
                     }
                 });
+
+                const datum = {
+                    result: evaluateResults(Object.values(telemetryDataCache), this.telemetry),
+                    ...latestTimestamp
+                };
+
                 return {
                     id: this.id,
-                    data: this.formatData(latestDatum, options.telemetryObjects)
+                    data: datum
```

Comment:
Do we not need to format the datum anymore?

Reply:
Format datum take each incoming subscription datum and updates the the class data cache and latest timestamp. We are doing the same just within the request itself so we aren't using formatData here in requests.

Reply:
Got it